core/src/main/java/org/apache/iceberg/Partitioning.java

```diff
@@ -177,4 +184,35 @@ public Void alwaysNull(int fieldId, String sourceName, int sourceId) {
       return null;
     }
   }
+
+  /**
+   * Builds a common partition type for all specs in a table.
```

Comment:
Could we add some detail in here? Seems like we are building a Struct which contains every field which has been part of a partitionspec in the past. So I would probably just write "Builds a struct containing all columns which have been ever been a part of a partition spec in this table"

Reply:
Added one more sentence below to clarify.